app/models/manageiq/providers/vmware/infra_manager/provision/cloning.rb

```diff
@@ -105,12 +105,14 @@ def start_clone(clone_options)
 
     [:transform, :config, :customization, :linked_clone].each { |key| vim_clone_options[key] = clone_options[key] }
 
-    [:folder, :host, :datastore, :pool, :snapshot].each do |key|
+    [:folder, :host, :pool, :snapshot].each do |key|
       ci = clone_options[key]
       next if ci.nil?
       vim_clone_options[key] = ci.ems_ref_obj
     end
 
+    vim_clone_options[:datastore] = clone_options[:host].host_storages.find_by(:storage_id => clone_options[:datastore].id).ems_ref
```

Comment:
@Fryguy @agrare Both `clone_options[:host]` and `clone_options[:datastore]` will be populated at this time as those fields are validated during the `placement` state-machine step since they are required properties to pass to VMware.